libs/designsystem/tab-navigation/src/tab-navigation/tab-navigation.component.ts

```diff
@@ -142,4 +134,29 @@ export class TabNavigationComponent implements AfterViewInit {
       });
     }
   }
+
+  private setTabElements() {
+    this.tabs.forEach((tab) => {
+      this.tabElements.push(tab.nativeElement);
+    });
+
+    this.tabElements.forEach((tabElement) =>
+      this.tabButtonElements.push(tabElement.querySelector('[role="tab"]'))
+    );
+
+    this.selectTab(this.selectedIndex);
+    this.focusTab(this.focusIndex);
+
+    setTimeout(() => {
+      this.scrollToTab(this.focusIndex);
+    }, this.DEBOUNCE_TIME_MS);
```

Comment:
Seems this has both been moved _and_ refactored (`selectTab` and `focusTab` called outside timeout) — are we sure this doesn't cause any regressions? 🤔

Reply:
The refactoring is intentional, otherwise there will be a visible delay when removing or inserting tab-items, as selection and focus happens asynchronously: 
https://github.com/kirbydesign/designsystem/assets/42470636/259c1807-f46e-49a8-b90a-4f4c9fc98867

Both `selectTab` and `focusTab` are all about manipulating the slotted html elements, and they are either called in AfterViewInit or when changes to a ContentChild is observed, so they should be safe to run at that point without delay.